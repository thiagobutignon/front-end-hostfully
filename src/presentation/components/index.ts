export { default as ErrorComponent } from './error/error'
export { default as HeaderComponent } from './header/header'
export { default as InputError } from './input/input-error'
export { default as InputWithLabel } from './input/input-with-label'
export { default as LayoutComponent } from './layout/layout'
export { default as PropertyInfoComponent } from './property-info/property-info'
export { default as PropertyCardComponent } from './select/property-select'
export { default as ThemeModeComponent } from './theme-mode/theme-mode'
