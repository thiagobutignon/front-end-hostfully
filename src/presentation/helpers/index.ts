export * from './populate-form-fields'
export * from './wrap-chakra'
