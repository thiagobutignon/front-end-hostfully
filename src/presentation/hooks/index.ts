export * from './booking'
export * from './error'
export * from './property'
