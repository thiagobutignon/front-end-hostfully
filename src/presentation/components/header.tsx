import { Box, Flex, Menu, Stack, Text } from '@chakra-ui/react'

import { ThemeModeComponent } from '@/presentation/components'

const HeaderComponent: React.FC = () => {
  return (
    <>
      <Box bg="reverseBackground" px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <Text color="reverseText" fontSize="xl" fontWeight={700}>
            Hostfully
          </Text>

          <Flex alignItems={'center'}>
            <Stack direction={'row'}>
              <Menu>
                <ThemeModeComponent />
              </Menu>
            </Stack>
          </Flex>
        </Flex>
      </Box>
    </>
  )
}

export default HeaderComponent
