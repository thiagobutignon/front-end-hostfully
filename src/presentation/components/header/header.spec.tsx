import { HeaderComponent } from '@/presentation/components'
import { render } from '@testing-library/react'

describe('HeaderComponent', () => {
  it('renders correctly', () => {
    const { getByText, getByTestId } = render(<HeaderComponent />)

    expect(getByText('Hostfully')).toBeInTheDocument()
    expect(getByTestId('mock-theme-mode-component')).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<HeaderComponent />)
    expect(asFragment()).toMatchSnapshot()
  })
})
