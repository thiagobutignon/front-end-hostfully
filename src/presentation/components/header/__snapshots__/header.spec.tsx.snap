// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-2uykgx"
    data-testid="mock-theme-mode-component"
  >
    <div
      class="css-q4hjtx"
    >
      <p
        class="chakra-text css-exm3rk"
      >
        Hostfully
      </p>
      <div
        class="css-70qvj9"
      >
        <div
          class="chakra-stack css-1f0wxn3"
        >
          <button
            class="chakra-button css-1nj3wf0"
            type="button"
          >
            <svg
              class="chakra-icon chakra-icon css-p00glj"
              data-testid="sun-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="5"
                />
                <path
                  d="M12 1v2"
                />
                <path
                  d="M12 21v2"
                />
                <path
                  d="M4.22 4.22l1.42 1.42"
                />
                <path
                  d="M18.36 18.36l1.42 1.42"
                />
                <path
                  d="M1 12h2"
                />
                <path
                  d="M21 12h2"
                />
                <path
                  d="M4.22 19.78l1.42-1.42"
                />
                <path
                  d="M18.36 5.64l1.42-1.42"
                />
              </g>
            </svg>
            Light
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
